core/state.go

```diff
@@ -25,11 +20,10 @@ type ErrMismatchedRoot struct {
 }
 
 func (e *ErrMismatchedRoot) Error() string {
-	newOld := " new "
 	if e.IsOld {
-		newOld = " old "
+		return fmt.Sprintf("state's current root: %s does not match state update's old root: %s", e.Got.String(), e.Want.String())
 	}
-	return fmt.Sprintf("mismatched %s root: want %s, got %s", newOld, e.Want.String(), e.Got.String())
+	return fmt.Sprintf("state's new root: %s does not match state update's new root: %s", e.Got.String(), e.Want.String())
```

Comment:
We should get rid of ErrMismatchedRoot, as per uber-go standards.

Reply:
done